examples/computer_vision/unets_tf_keras/model_def.py

```diff
@@ -61,8 +63,12 @@ def download_weights(self):
 
         # Use a file lock so only one worker on each node does the download
         with filelock.FileLock(os.path.join(weights_dir, "download.lock")):
-            urllib.request.urlretrieve(mobilenet_link, weights_dir + data_file)
-        return weights_dir + data_file
+            full_weights_path = weights_dir + data_file
+            if not os.path.exists(full_weights_path):
+                with tempfile.NamedTemporaryFile(delete=False) as ntf:
+                    urllib.request.urlretrieve(mobilenet_link, ntf.name)
+                    os.rename(ntf.name, full_weights_path)
```

Comment:
Since you can't rename across filesystem boundaries, I don't think this is a safe approach.

But a simpler approach which is extremely unlikely to hit a filesystem boundary issue would be:
```python
if not os.path.exists(full_weights_path):
  urllib.request.urlretrieve(mobilenet_link, full_weights_path + ".part")
  os.rename(full_weights_path + ".part", full_weights_path)
```

Reply:
ok makes sense